### PR TITLE
Issue 51521: Add switch to disable Server HTTP header

### PR DESCRIPTION
**Pull request by labkey-jeckels**
#### Rationale
Some organizations don't like having the product and version identified in HTTP headers. We can make it optional.

#### Related Pull Requests
- https://github.com/LabKey/platform/pull/5989

#### Changes
- Test the presence and absence of the header when toggling the setting

### CODE REVIEW

File: src/org/labkey/test/tests/AdminConsoleTest.java [hunk 1]
```diff
@@ -51,6 +59,74 @@ public String getProjectName()
...
+            GetServerHeaderCommand command = new GetServerHeaderCommand();
+            command.execute(cn, "/home");
+            return command._server;
+        }
+        catch (IOException | CommandException e)
+        {
+            throw new RuntimeException("Failed to get Server HTTP response header", e);
+        }
+    }
+        @Test
```
Comment (labkey-tchad) [inline src/org/labkey/test/tests/AdminConsoleTest.java]: Spacing looks messed up
```suggestion
    }
    
    @Test
```

File: src/org/labkey/test/tests/AdminConsoleTest.java [hunk 2]
```diff
@@ -51,6 +59,74 @@ public String getProjectName()
     }
 
     @Test
+    public void testServerHttpHeaderSetting()
+    {
+        goToAdminConsole().clickSiteSettings();
+        waitForElement(Locator.name("includeServerHttpHeader"));
+        WebElement checkbox = Locator.checkboxByName("includeServerHttpHeader").findElement(getDriver());
+
+        boolean originalValue = "true".equals(checkbox.getAttribute("checked"));
```
Comment (labkey-tchad) [inline src/org/labkey/test/tests/AdminConsoleTest.java]: "checked" is a valueless attribute. It's mere existence indicates that the checkbox is checked, even with `checked="false"`. This code implies that the `includeServerHttpHeader` checkbox is not written correctly.

That being said, Selenium understands standard HTML checkboxes.
```suggestion
        boolean originalValue = checkbox.isSelected());
```

**Review by labkey-tchad — CHANGES_REQUESTED** [review]

File: src/org/labkey/test/tests/AdminConsoleTest.java (hunk 2 above)
Comment (labkey-jeckels) [reply src/org/labkey/test/tests/AdminConsoleTest.java]: > "checked" is a valueless attribute. It's mere existence indicates that the checkbox is checked, even with `checked="false"`. This code implies that the `includeServerHttpHeader` checkbox is not written correctly.
> 
> That being said, Selenium understands standard HTML checkboxes.

Thanks Trey. I also fixed up the other code that I patterned my change after.

**Review by labkey-tchad — CHANGES_REQUESTED** [review]
Sorry I didn't catch this the first time.

File: src/org/labkey/test/tests/AdminConsoleTest.java [hunk 3]
```diff
@@ -50,6 +55,75 @@ public String getProjectName()
...
+        goToAdminConsole().clickSiteSettings();
+        waitForElement(Locator.name("includeServerHttpHeader"));
+        WebElement checkbox = Locator.checkboxByName("includeServerHttpHeader").findElement(getDriver());
+
+        boolean originalValue = checkbox.isSelected();
+
+        // Try with the setting on
+        if (!originalValue)
+            click(Locator.checkboxByName("includeServerHttpHeader"));
+        clickButton("Save");
```
Comment (labkey-tchad) [inline src/org/labkey/test/tests/AdminConsoleTest.java]: This checkbox should be manipulated via `CustomizeSitePage`.
```suggestion
        CustomizeSitePage customizeSitePage = goToAdminConsole().clickSiteSettings();
        boolean originalValue = customizeSitePage.isServerHttpHeaderEnabled();

        // Try with the setting on
        if (!originalValue)
            customizeSitePage.enableServerHttpHeader(true);
        customizeSitePage.save();
```

**Review by labkey-tchad — CHANGES_REQUESTED** [review]

File: src/org/labkey/test/tests/AdminConsoleTest.java (hunk 3 above)
Comment (labkey-jeckels) [reply src/org/labkey/test/tests/AdminConsoleTest.java]: OK. Made another update, including another improvement to the section I used as a reference.

**Review by labkey-tchad — APPROVED** [review]

**Pull request merged**